pnr/symbiflow-vtr-optimized/run-sf.sh

```diff
@@ -0,0 +1,42 @@
+# Run VPR on an example .eblif to collect PGO data
+
+pushd ${BUILD_ROOT}/symbiflow
+
+# Runs the following command, printing only lines starting
+# with '#' unless the command fails, in which case it will
+# print the entire log.
+quiet() {
+    ("$@" | tee full.log | grep -e '^#') || cat full.log
+}
+
+# Parameters
+NET=ibex_arty
+DEVICE=xc7a50t_test
+PART=xc7a35tcsg324-1
+EBLIF=benchmarks/circuits/${NET}.eblif
+SDC=benchmarks/sdc/${NET}.sdc
+PLACE_CONSTRAINTS=benchmarks/place_constr/${NET}.place
+PATH=${BUILD_ROOT}/build/vpr:${BUILD_ROOT}/build/utils/fasm:${BUILD_ROOT}/symbiflow/bin:${PATH}
+
+# Pack
+quiet symbiflow_pack -e ${EBLIF} -d ${DEVICE} -s ${SDC}
+
+# Place
+# This should be:
```

Comment:
I'm not sure I understand this comment?  Can you explain what this working around?

Reply:
Otherwise, `symbiflow_generate_constraints` in `symbiflow_place` fails with:

```
Traceback (most recent call last):
  File ".../symbiflow/share/symbiflow/scripts/prjxray_create_ioplace.py", line 178, in <module>
    main()
  File ".../symbiflow/share/symbiflow/scripts/prjxray_create_ioplace.py", line 169, in main
    io_place.output_io_place(args.output)
  File ".../symbiflow/share/symbiflow/scripts/vpr_io_place.py", line 151, in output_io_place
    max_name_length = max(len(c.name) for c in self.constraints.values())
```

Reply:
I think you failed to copy the whole exception?

Reply:
Yes, sorry:

    ValueError: max() arg is an empty sequence

Reply:
This also allows me to remove the dependency on `python-constraint` which requires the `conda-forge` channel.

Reply:
> This also allows me to remove the dependency on `python-constraint` which requires the `conda-forge` channel.

I believe you can get `python-constraint` from `PyPi`?

Reply:
> Yes, sorry:
> 
> ```
> ValueError: max() arg is an empty sequence
> ```

So couple thoughts.  First, this error probably deserves to be fixed in upstream arch-defs, so it would be good to file an issue with replication instructions.  Second, I believe the fact that you hit this error likely indicates something else is wrong, not sure what.

Reply:
https://github.com/SymbiFlow/symbiflow-arch-defs/issues/1881

I think I might have found the problem:
```
ls -l benchmarks/sdc/ibex_arty.sdc
lrwxrwxrwx 1 dusty users 94 Dec 10 06:07 benchmarks/sdc/ibex_arty.sdc -> /tmpfs/src/github/symbiflow-arch-defs-continuous-install/xc/xc7/tests/soc/ibex/pins_artya7.sdc
```

Bad link.

Reply:
I think @acomodi handles this in the VTR / symbiflow test stuff?

Reply:
https://github.com/SymbiFlow/symbiflow-arch-defs/issues/1882

Reply:
`vpr_io_place.py` still fails. If I understand correctly, the `ibex_arty.eblif` just doesn't have any constraints:

```
$ grep LOC_PAIRS symbiflow/benchmarks/circuits/ibex_arty.eblif
.param IO_LOC_PAIRS "NONE"
.param IO_LOC_PAIRS "NONE"
.param IO_LOC_PAIRS "NONE"
.param IO_LOC_PAIRS "NONE"
.param IO_LOC_PAIRS "NONE"
.param IO_LOC_PAIRS "NONE"
```

Reply:
The SDC issue needs to be solved in arch-defs (https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/1564#discussion_r538254173).

Regarding the IO constraints, given that the benchmarks package was originally generated to run in the VTR flow, they are already expressed as VPR placement constraints without in a `.place` file. The ibex test still uses the PCF+SDC format for constraints, hence the .eblif does not contain the IO_LOC_PAIRS.
I believe that it is worth switching to another test (e.g. picosoc or minilitex) in this PR rather then ibex, until at least we get to transition to the XDC format for all the tests that are going to be installed in the benchmark package.

Reply:
I believe picosoc is too small to get sufficient profile data. I will try minilitex.

Reply:
minilitex should work:
```
symbiflow:                                 arch.timing.xml/minilitex_ddr_arty.eblif/common                                            		OK (took 526.96 seconds)
```
from @acomodi VTR + Symbiflow test on upstream: 
https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/1564
https://source.cloud.google.com/results/invocations/670bdd9e-04eb-4f0c-a667-b13421f1070d/log

Reply:
I've switched to the `minilitex_ddr_arty` design for profiling, and removed the workaround.